### PR TITLE
hdfview: 2.14 -> 3.1.2

### PR DESCRIPTION
**Pull request by astro**
###### Motivation for this change

Major version update

Open questions:
* hdfview was the only package depending on hdf_java. Remove it?
* Enable hdf4/hdf5's `javaSupport` by default?

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
> * hdfview was the only package depending on hdf_java. Remove it?

It is hdfview specific so I think it should be safe to remove. We can always revert the removal if it is still needed.

> Enable hdf4/hdf5's javaSupport by default?

Is that something you want by default and is used in your daily workflow then yes. If not I think we can leave it disabled because java blows up the dependency chain quite a bit.

File: pkgs/tools/misc/hdfview/default.nix [hunk 1]
```diff
@@ -1,39 +1,50 @@
...
-      --replace "@JAVABIN@" "${javac}/bin/" \
-      --replace "@INSTALLDIR@" "$out"
+    mkdir -p $out/bin $out/lib
+    cp -a build/dist/HDFView/bin/HDFView $out/bin/
+    cp -a build/dist/HDFView/lib/app $out/lib/
+    ln -s ${jdk}/lib/openjdk $out/lib/runtime
+
+    mkdir -p $out/share/applications $out/share/icons/hicolor/32x32/apps
+    cp src/HDFView.png $out/share/icons/hicolor/32x32/apps/
+    cp $desktopItem/share/applications/* $out/share/applications/
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/hdfview/default.nix:41]: This could be replaced with copyDesktopItems.

File: pkgs/tools/misc/hdf4/default.nix [hunk 2]
```diff
@@ -52,7 +56,9 @@ stdenv.mkDerivation rec {
     libjpeg
     szip
     zlib
-  ] ++ lib.optionals uselibtirpc [
+  ]
+  ++ lib.optional javaSupport jdk
+  ++ lib.optionals uselibtirpc [
     libtirpc
   ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/hdf4/default.nix:63]:
```suggestion
  ] ++ lib.optional javaSupport jdk
  ++ lib.optional uselibtirpc libtirpc;
```

File: pkgs/tools/misc/hdf4/default.nix [hunk 3]
```diff
@@ -7,8 +7,12 @@
 , libtirpc
 , zlib
 , szip ? null
+, javaSupport ? false
+, jdk
 }:
-let uselibtirpc = stdenv.isLinux;
+let
+  uselibtirpc = stdenv.isLinux;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/hdf4/default.nix:14]:
```suggestion
```

File: pkgs/tools/misc/hdf4/default.nix [hunk 4]
```diff
@@ -7,8 +7,12 @@
 , libtirpc
 , zlib
 , szip ? null
+, javaSupport ? false
+, jdk
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/hdf4/default.nix:11]:
```suggestion
, jdk
, uselibtirpc ? stdenv.isLinux
```

**Comment by astro** [comment]
Thank you @SuperSandro2000. I repushed with all your suggestions.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113349` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>hdf5</li>
<li>hdfview</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
hdf5:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/tools/misc/hdf5/default.nix:67:3:

```
   |
67 |   meta = {
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/tools/misc/hdf5/default.nix:56:5:

```
   |
56 |     ./bin-mv.patch
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
</details>
hdfview:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/tools/misc/hdfview/default.nix:44:3:

```
   |
44 |   meta = {
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/tools/misc/hdfview/default.nix:21:3:

```
   |
21 |   buildPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/misc/hdfview/default.nix:34:3:

```
   |
34 |   installPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
</details>

**Comment by astro** [comment]
Regarding nixpkgs-review-checks:
* I do not want to become the hdfview maintainer unless strictly necessary. There was none before.
* I do not know how to document the preexisting hdf5 patch.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113349` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>hdf5</li>
<li>hdfview</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
hdf5:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/tools/misc/hdf5/default.nix:67:3:

```
   |
67 |   meta = {
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/tools/misc/hdf5/default.nix:56:5:

```
   |
56 |     ./bin-mv.patch
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
</details>
hdfview:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/tools/misc/hdfview/default.nix:44:3:

```
   |
44 |   meta = {
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/tools/misc/hdfview/default.nix:21:3:

```
   |
21 |   buildPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/misc/hdfview/default.nix:34:3:

```
   |
34 |   installPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
</details>

**Comment by SuperSandro2000** [comment]
> I do not want to become the hdfview maintainer unless strictly necessary. There was none before.

That is alright. We added that to make people aware that packages have no maintainer and that they might pick it up if the package is important for them.

> I do not know how to document the preexisting hdf5 patch.

Thats alright, too.

**Comment by jbqubit** [comment]
Thank you @astro !

**Pull request merged**